cwms-data-api/src/test/java/cwms/cda/api/BasinControllerTestIT.java

```diff
@@ -1,59 +1,339 @@
 package cwms.cda.api;
 
+import cwms.cda.api.enums.Nation;
+import cwms.cda.data.dao.LocationsDaoImpl;
+import cwms.cda.data.dto.Location;
+import cwms.cda.data.dto.LocationIdentifier;
+import cwms.cda.data.dto.basin.Basin;
 import cwms.cda.formatters.Formats;
+import fixtures.CwmsDataApiSetupCallback;
+import fixtures.TestAccounts;
 import io.restassured.filter.log.LogDetail;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.EnumSource;
 
+import static cwms.cda.api.Controllers.FAIL_IF_EXISTS;
+import static cwms.cda.security.KeyAccessManager.AUTH_HEADER;
+
 import javax.servlet.http.HttpServletResponse;
 
-import java.net.URLEncoder;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
 
+import static cwms.cda.data.dao.DaoTest.getDslContext;
 import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
 @Tag("integration")
 class BasinControllerTestIT extends DataApiTestIT
 {
 	private static final String SPK = "SPK";
+	private static final Basin BASIN;
+	private static final String DELETE_ACTION = "DELETE ALL";
+	static {
+		try {
+			BASIN = new Basin.Builder()
+					.withBasinId(new LocationIdentifier.Builder()
+							.withName("TEST_LOCATION")
+							.withOfficeId(SPK)
+							.build())
+					.withSortOrder(1.0)
+					.withTotalDrainageArea(1000.0)
+					.withContributingDrainageArea(500.0)
+					.withAreaUnit("m2")
+					.build();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@BeforeAll
+	public static void setup() throws Exception {
+		CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+		databaseLink.connection(c -> {
+			DSLContext ctx = getDslContext(c, databaseLink.getOfficeId());
```

Comment:
Use the defined office ID here, and pass "CWMS_20" in as second parameter to the ::connection function.

@rma-bryson @rma-psmorris @adamkorynta @rma-rripken @RyanM-RMA 
FYI, When the integration tests run with a fresh database it's created with the primary office as "HQ", which is what's done for shared offices. Sorry, I should've noticed that in the other PRs and did not, plus my main testing is also a curated single office database.

I'm going to spend some time next week on getting the integration tests to run on teamcity or github (almost have the db running on the newer "oracle free" database.)

Reply:
Also it is acceptable to use an arbitrary office, it just needs to be consistently used through the test and you may need to go into the database init scripts in the resource directory to make sure permissions are assigned. If fact sense our short term goal is supporting multi office I encourage all of us to use a different office at least per test.